client/Components/SingleCohort/SingleCohort.js

```diff
@@ -0,0 +1,47 @@
+import React, { useEffect } from 'react'
+import { connect } from 'react-redux'
+import { fetchCohort, fetchStudents, fetchStretches } from '../../store/thunks'
+import Sidebar from '../Sidebar'
+
+const mapStateToProps = ({ cohort, students, stretches }) => {
+  return {
+    cohort,
+    students,
+    stretches
+  }
+}
+
+const mapDispatchToProps = dispatch => {
+  return {
+    fetchCohort: cohortId => dispatch(fetchCohort(cohortId)),
```

Comment:
wondering if this is case of taking business logic in back-end motto too far? Think in this case probably okay to filter cohorts array in state to get current cohort than make another API call to get data that is already in state

Reply:
Yeah - so, the cohorts (all of them) are on state I think b/c the adminHomeView fetches them. I'm find just filtering, for sure.

Reply:
So one thing about this - if you load the page without first going to AdminHomeView, it means there are no cohorts on state. Is it worth keeping fetchCohort just as a defense to that?

Reply:
Good point. So to defend against that while not doing an unneeded API call, can do IF statement so fetchCohorts + filtering only invoked if reducer is empty